k8s/upgrade-job/src/events/event_recorder.rs

```diff
@@ -159,12 +160,39 @@ impl EventRecorderBuilder {
             resource_version: None,
         };
 
+        let (tx, mut rx) = mpsc::channel::<Event>(10);
+
+        let aborter = tokio::spawn(async move {
+            let event_handler =
+                Recorder::new(k8s_client.client(), reporter_name.into(), job_obj_ref);
+
+            // Function exits when 'None'. Avoids panics.
+            let mut current_event = rx.recv().await;
+
+            while let Some(event) = &current_event {
+                // Hacky clone for the Event.
+                let event = Event {
+                    type_: event.type_,
+                    reason: event.reason.clone(),
+                    note: event.note.clone(),
+                    action: event.action.clone(),
+                    secondary: event.secondary.clone(),
+                };
+                if let Err(error) = event_handler.publish(event).await.context(EventPublish) {
```

Comment:
Should we have a queue of events that need to be published? Otherwise if there's an intermittent failure we'll miss events. If we keep a queue of failed publishes, we retry them?

Reply:
I'm planning on including this in a future release, but not this one.